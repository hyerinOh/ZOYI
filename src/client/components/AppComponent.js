import React, { Component } from "react";
import "../assets/styles/app.css";
import MainPage from "./MainPage";

class AppComponent extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    console.log("Redux props", this.props);
    return (
      <div>
        <MainPage {...this.props} />
      </div>
    );
  }
}

export default AppComponent;
