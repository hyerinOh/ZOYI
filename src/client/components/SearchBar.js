import React, { Component } from "react";
import "../assets/styles/app.css";
import ShowCountryLists from "./ShowCountryLists";
import axios from "axios";

export default class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      timerID: null,
      index: 0,
      isDone: false,
      throttleCheck: 0
    };
  }

  componentDidMount() {
    window.addEventListener("scroll", this.handleScroll);
  }

  componentWillUnmount() {
    window.removeEventListener("scroll", this.handleScroll);
  }

  handleScroll = () => {
    const { innerHeight } = window;
    const { scrollHeight } = document.body;
    const { scrollY } = window;
    // throttling
    this.setState({
      throttleCheck: setTimeout(() => {
        if (this.state.index >= this.props.countryLists.length) {
          this.setState({
            isDone: true
          });
        }
        if (
          scrollHeight - (innerHeight + scrollY) < 100 &&
          !this.state.isDone
        ) {
          if (this.state.throttleCheck) {
            this.updateList();
            this.setState({
              throttleCheck: 0
            });
          }
        }
      }, 500)
    });
  };

  updateList = () => {
    this.setState({
      index: this.state.index + 5
    });
  };

  getCountryLists = () => {
    if (this.props.searchValue) {
      const params = ["name", "alpha2Code", "callingcode", "capital", "region"];
      params.map(param => {
        axios
          .get(
            `https://restcountries.eu/rest/v2/${param}/${
              this.props.searchValue
            }?fields=alpha2Code;capital;name;region;callingCodes`
          )
          .then(response => {
            console.log("response", response.data);
            this.props.saveCountryList(response.data);
            this.updateList();
          })
          .catch(error => {
            console.log(error);
          });
      });
    } else {
      axios
        .get(
          "https://restcountries.eu/rest/v2/all?fields=alpha2Code;capital;name;region;callingCodes"
        )
        .then(response => {
          this.props.saveCountryList(response.data);
          this.updateList();
        })
        .catch(error => {
          console.log(error);
        });
    }
  };

  handleChange = ev => {
    const { searchCountry } = this.props;
    ev.preventDefault();
    searchCountry(ev.target.value);
    console.log("pppp", this.props);

    // debouncing
    if (this.state.timerID) {
      this.setState({
        timerID: clearTimeout(this.state.timerID)
      });
    }

    this.setState({
      timerID: setTimeout(() => {
        this.getCountryLists();
      }, 1000)
    });
  };

  render() {
    return (
      <div>
        <form
          className="searchEngineWrapper"
          onSubmit={ev => ev.preventDefault()}
        >
          <input
            type="text"
            className="searchBar"
            placeholder="나라정보 검색"
            value={this.props.searchValue}
            onChange={this.handleChange}
          />
        </form>
        <div className="listWrapper">
          <ShowCountryLists {...this.props} index={this.state.index} />
        </div>
      </div>
    );
  }
}
