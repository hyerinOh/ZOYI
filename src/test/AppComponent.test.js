import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import { shallow, mount } from "enzyme";
import AppComponent from "../client/components/AppComponent";
import SearchBar from "../client/components/SearchBar";
configure({ adapter: new Adapter() });

const initialState = {
  searchValue: "asia",
  countryNameSorting: "asc",
  countryCodeSorting: "asc",
  countryCallingCodeSorting: "asc",
  countryCapitalSorting: "asc",
  countryRegionSorting: "asc",
  countryLists: [
    {
      name: "Korea",
      alpha2Code: "KR",
      callingCoes: ["82"],
      capital: "seoul",
      region: "Asia"
    }
  ]
};

describe("<AppComponent />", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<AppComponent {...initialState} />);
  });

  it("should render a one <div>", () => {
    expect(wrapper.find("div.mainWrapper").to.have.lengthOf(1));
  });
});
