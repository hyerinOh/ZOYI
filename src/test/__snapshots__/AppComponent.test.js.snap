// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent matches snapshot 1`] = `null`;
